// konsta/react/components/App.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useRef } from 'react';
import { KonstaProvider } from '../shared/KonstaProvider.js';
import { useAutoTheme } from '../shared/use-auto-theme.js';
import { AppClasses } from '../../shared/classes/AppClasses.js';
const App = props => {
  const {
    component = 'div',
    className,
    theme = 'material',
    dark = true,
    materialTouchRipple = true,
    iosHoverHighlight = true,
    safeAreas = true,
    ref,
    // Children
    children,
    // Rest
    ...rest
  } = props;
  const elRef = useRef(null);
  const Component = component;
  const attrs = {
    ...rest
  };
  const currentTheme = useAutoTheme(theme);
  const classes = AppClasses({
    ...props,
    safeAreas
  }, currentTheme, className);
  return /*#__PURE__*/React.createElement(KonstaProvider, {
    theme: currentTheme,
    dark: dark,
    materialTouchRipple: materialTouchRipple,
    iosHoverHighlight: iosHoverHighlight,
    autoThemeDetection: false
  }, /*#__PURE__*/React.createElement(Component, _extends({
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    className: classes
  }, attrs), children));
};
App.displayName = 'App';
export default App;
