// konsta/react/components/DialogButton.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useRef } from 'react';
import { useTheme } from '../shared/use-theme.js';
import Button from './Button.js';
const DialogButton = props => {
  const {
    component = 'button',
    className,
    ios,
    material,
    disabled,
    strong,
    strongIos,
    strongMaterial,
    // Children
    children,
    ref,
    // Rest
    ...rest
  } = props;
  const elRef = useRef(null);
  const attrs = {
    ...rest
  };
  const theme = useTheme({
    ios,
    material
  });
  const isStrong = typeof strong === 'undefined' ? theme === 'ios' ? strongIos : strongMaterial : strong;
  if (theme === 'ios') return /*#__PURE__*/React.createElement(Button, _extends({
    component: component,
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    large: true,
    rounded: true,
    tonal: !isStrong,
    className: className,
    disabled: disabled
  }, attrs), children);
  return /*#__PURE__*/React.createElement(Button, _extends({
    component: component,
    inline: true,
    rounded: true,
    disabled: disabled,
    clear: !isStrong,
    className: className,
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    }
  }, attrs), children);
};
DialogButton.displayName = 'DialogButton';
export default DialogButton;
