// konsta/react/components/Fab.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useRef } from 'react';
import { useTheme } from '../shared/use-theme.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { useTouchRipple } from '../shared/use-touch-ripple.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
import { FabClasses } from '../../shared/classes/FabClasses.js';
import { FabColors } from '../../shared/colors/FabColors.js';
import Glass from './Glass.js';
const Fab = props => {
  const {
    component = 'a',
    className,
    colors: colorsProp,
    href,
    text,
    textPosition = 'after',
    icon,
    ios,
    material,
    // Children
    children,
    ref,
    // Rest
    ...rest
  } = props;
  const rippleElRef = useRef(null);
  const Component = component;
  const attrs = {
    ...rest
  };
  const theme = useTheme({
    ios,
    material
  });
  const themeClasses = useThemeClasses({
    ios,
    material
  });
  const dark = useDarkClasses();
  useTouchRipple(rippleElRef, theme === 'material');
  const colors = FabColors(colorsProp, dark);
  const c = themeClasses(FabClasses(props, colors, dark), className);
  return /*#__PURE__*/React.createElement(Glass, _extends({
    component: Component,
    className: text ? c.base.withText : c.base.iconOnly,
    href: href,
    ref: el => {
      rippleElRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    role: "button",
    tabIndex: "0"
  }, attrs), text && textPosition === 'before' && /*#__PURE__*/React.createElement("span", {
    className: c.text
  }, text), icon && /*#__PURE__*/React.createElement("span", {
    className: c.icon
  }, icon), text && textPosition === 'after' && /*#__PURE__*/React.createElement("span", {
    className: c.text
  }, text), children);
};
Fab.displayName = 'Fab';
export default Fab;
