// konsta/react/components/Notification.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useRef } from 'react';
import { NotificationsClasses } from '../../shared/classes/NotificationsClasses.js';
import { NotificationsColors } from '../../shared/colors/NotificationsColors.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
import { useTheme } from '../shared/use-theme.js';
import DeleteIcon from './icons/DeleteIcon.js';
import Glass from './Glass.js';
const Notification = props => {
  const {
    component = 'div',
    className,
    colors: colorsProp,
    button,
    icon,
    title,
    titleRightText,
    subtitle,
    text,
    opened,
    onClose,
    ios,
    material,
    // Children
    children,
    ref,
    // Rest
    ...rest
  } = props;
  const elRef = useRef(null);
  const Component = component;
  const attrs = {
    ...rest
  };
  const theme = useTheme({
    ios,
    material
  });
  const themeClasses = useThemeClasses({
    ios,
    material
  });
  const dark = useDarkClasses();
  const colors = NotificationsColors(colorsProp, dark);
  const c = themeClasses(NotificationsClasses({
    ...props
  }, colors), className);
  if (theme === 'ios') return /*#__PURE__*/React.createElement(Glass, _extends({
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    component: component,
    className: c.base
  }, attrs), icon && /*#__PURE__*/React.createElement("div", {
    className: c.icon
  }, icon), /*#__PURE__*/React.createElement("div", {
    className: c.content
  }, /*#__PURE__*/React.createElement("div", {
    className: c.header
  }, title && /*#__PURE__*/React.createElement("div", {
    className: c.title
  }, title), titleRightText && /*#__PURE__*/React.createElement("div", {
    className: c.titleRightText
  }, titleRightText), button && /*#__PURE__*/React.createElement("div", {
    className: c.button,
    role: "button",
    tabIndex: "0",
    onClick: onClose
  }, /*#__PURE__*/React.createElement(DeleteIcon, {
    theme: theme,
    className: c.deleteIcon
  }))), subtitle && /*#__PURE__*/React.createElement("div", {
    className: c.subtitle
  }, subtitle), text && /*#__PURE__*/React.createElement("div", {
    className: c.text
  }, text), children));
  return /*#__PURE__*/React.createElement(Component, _extends({
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    className: c.base
  }, attrs), /*#__PURE__*/React.createElement("div", {
    className: c.header
  }, icon && /*#__PURE__*/React.createElement("div", {
    className: c.icon
  }, icon), /*#__PURE__*/React.createElement("div", {
    className: c.contentWrapper
  }, /*#__PURE__*/React.createElement("div", {
    className: c.contentTitle
  }, title && /*#__PURE__*/React.createElement("div", {
    className: c.title
  }, title), titleRightText && /*#__PURE__*/React.createElement("div", {
    className: c.titleRightText
  }, titleRightText)), /*#__PURE__*/React.createElement("div", {
    className: c.content
  }, subtitle && /*#__PURE__*/React.createElement("div", {
    className: c.subtitle
  }, subtitle), text && /*#__PURE__*/React.createElement("div", {
    className: c.text
  }, text), children)), button && /*#__PURE__*/React.createElement("div", {
    className: c.button,
    role: "button",
    tabIndex: "0",
    onClick: onClose
  }, /*#__PURE__*/React.createElement(DeleteIcon, {
    theme: theme,
    className: c.deleteIcon
  }))));
};
Notification.displayName = 'Notification';
export default Notification;
