// konsta/react/components/Panel.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useRef } from 'react';
import { cls } from '../../shared/cls.js';
import { PanelClasses } from '../../shared/classes/PanelClasses.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { PanelColors } from '../../shared/colors/PanelColors.js';
import Glass from './Glass.js';
import { useTheme } from '../shared/use-theme.js';
const Panel = props => {
  const {
    component = 'div',
    className,
    colors: colorsProp,
    side = 'left',
    opened,
    backdrop = true,
    floating = false,
    onBackdropClick,
    ios,
    material,
    // Children
    children,
    ref,
    // Rest
    ...rest
  } = props;
  const elRef = useRef(null);
  const state = opened ? 'opened' : 'closed';
  const Component = component;
  const attrs = {
    ...rest
  };
  const theme = useTheme({
    ios,
    material
  });
  const themeClasses = useThemeClasses({
    ios,
    material
  });
  const dark = useDarkClasses();
  const colors = PanelColors(colorsProp, dark);
  const c = themeClasses(PanelClasses({
    ...props,
    floating
  }, colors), className);
  const classes = cls(c.base, c[side][state]);
  const setRef = el => {
    elRef.current = el;
    if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, backdrop && /*#__PURE__*/React.createElement("div", {
    className: c.backdrop[state],
    onClick: onBackdropClick
  }), theme === 'ios' && floating ? /*#__PURE__*/React.createElement(Glass, _extends({
    component: component,
    ref: setRef,
    className: classes,
    highlight: false
  }, attrs), children) : /*#__PURE__*/React.createElement(Component, _extends({
    ref: setRef,
    className: classes
  }, attrs), children));
};
Panel.displayName = 'Panel';
export default Panel;
