// konsta/react/components/SegmentedButton.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useRef } from 'react';
import Button from './Button.js';
import { useSegmentedContext } from './SegmentedContext.js';
const SegmentedButton = props => {
  const {
    active,
    children,
    outline,
    strong: buttonStrong,
    rounded: buttonRounded,
    raised: buttonRaised,
    ref,
    ...rest
  } = props;
  const elRef = useRef(null);
  const {
    strong,
    rounded
  } = useSegmentedContext();
  return /*#__PURE__*/React.createElement(Button, _extends({
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    segmented: true,
    segmentedActive: active,
    segmentedStrong: strong,
    rounded: rounded && strong
  }, rest), children);
};
SegmentedButton.displayName = 'SegmentedButton';
export default SegmentedButton;
