// konsta/react/components/ToolbarPane.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { useEffect, useRef, useState } from 'react';
import { useTheme } from '../shared/use-theme.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { ToolbarPaneClasses } from '../../shared/classes/ToolbarPaneClasses.js';
import Glass from './Glass.js';
import { useToolbarContext } from './ToolbarContext.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
import { ToolbarPaneColors } from '../../shared/colors/ToolbarPaneColors.js';
import { useIosTabbarHighlight } from '../../shared/use-ios-tabbar-highlight.js';
const ToolbarPane = props => {
  const {
    component = 'div',
    className,
    colors: colorsProp,
    ios,
    material,
    children,
    ref,
    ...rest
  } = props;
  const elRef = useRef(null);
  const highlightElRef = useRef(null);
  const [highlightStyle, setHighlightStyle] = useState({
    transform: '',
    width: ''
  });
  const highlightData = useRef({});
  const highlightInnerRef = useRef(null);
  const highlightThumbRef = useRef(null);
  const [hasTabbarLinks, setHasTabbarLinks] = useState(false);
  const themeClasses = useThemeClasses({
    ios,
    material
  });
  const attrs = {
    ...rest
  };
  const theme = useTheme({
    ios,
    material
  });
  const {
    tabbar
  } = useToolbarContext();
  const hasHighlight = theme === 'ios' && tabbar;
  const dark = useDarkClasses();
  const colors = ToolbarPaneColors(colorsProp, dark);
  const c = themeClasses(ToolbarPaneClasses({
    ...props,
    tabbar
  }, colors, dark), className);
  const {
    attachEvents,
    detachEvents,
    onUpdated
  } = useIosTabbarHighlight({
    getEl: () => elRef.current,
    getHighlightEl: () => highlightElRef.current,
    getHighlightInnerEl: () => highlightInnerRef.current,
    getHighlightThumbEl: () => highlightThumbRef.current,
    getHasHighlight: () => hasHighlight,
    data: highlightData.current,
    getHasTabbarLinks: () => hasTabbarLinks,
    setHasTabbarLinks,
    classes: c,
    getHighlightStyle: () => highlightStyle,
    setHighlightStyle
  });
  useEffect(() => {
    onUpdated();
  }, [children]);
  useEffect(() => {
    attachEvents();
    return () => {
      detachEvents();
    };
  });
  if (theme === 'material') return children;
  return /*#__PURE__*/React.createElement(Glass, _extends({
    component: component,
    ref: el => {
      elRef.current = el;
      if (ref && typeof ref === 'function') ref(el);else if (ref) ref.current = el;
    },
    className: c.base
  }, attrs), children, hasHighlight && /*#__PURE__*/React.createElement("span", {
    className: c.highlight,
    style: highlightStyle,
    ref: highlightElRef
  }, /*#__PURE__*/React.createElement("span", {
    ref: highlightInnerRef,
    className: c.highlightInner
  }), /*#__PURE__*/React.createElement("span", {
    ref: highlightThumbRef,
    className: c.highlightThumb
  })));
};
ToolbarPane.displayName = 'ToolbarPane';
export default ToolbarPane;
